innopoints/models/activity.py

```diff
@@ -36,7 +36,13 @@ class Activity(db.Model):
     """Represents a volunteering activity in the project."""
     __tablename__ = 'activities'
     __table_args__ = (
-        db.CheckConstraint(f'(fixed_reward AND working_hours = 1) '
+        db.CheckConstraint('working_hours == NULL OR working_hours >= 0',
+                           name='working hours are non-negative'),
+        db.CheckConstraint('people_required == NULL OR people_required > 0',
+                           name='people required are unset or positive'),
+        db.CheckConstraint('NOT draft AND working_hours != NULL',
```

Comment:
Did you mean `OR` instead of `AND` here? The check will fail if the activity is a draft, regardless of the value in `working_hours`